infer/src/absint/ProcCfg.ml

```diff
@@ -201,6 +201,100 @@ module Normal = struct
   let wto = Procdesc.get_wto
 end
 
+(** Forward CFG with exceptional control-flow for throw exception instruction only*)
+module ExceptionalThrowOnly = struct
+  module Node = DefaultNode
+
+  type id_node_map = Node.t list Procdesc.IdMap.t
+
+  type t = Procdesc.t * id_node_map
+
+  type instrs_dir = Instrs.not_reversed
+
+  let instrs = Procdesc.Node.get_instrs
+
+  let fold_normal_succs _ n ~init ~f = n |> Procdesc.Node.get_succs |> List.fold ~init ~f
+
+  let fold_normal_preds _ n ~init ~f = n |> Procdesc.Node.get_preds |> List.fold ~init ~f
+  
+  let fold_exceptional_succs _ n ~init ~f = n |> Procdesc.Node.get_exn |> List.fold ~init ~f
+
+  let fold_exceptional_preds (_, exn_pred_map) n ~init ~f =
+    match Procdesc.IdMap.find (Procdesc.Node.get_id n) exn_pred_map with
+    | exn_preds ->
+        List.fold exn_preds ~init ~f
+    | exception Caml.Not_found ->
+        init
+
+  (** we fold the exception flow only when the throw instruction is encountered and exn is returned. 
+  Under this circumstances, resources could be disposed in finaly black through exception sink node *)
+  let fold_normal_or_exn_succs fold_normal_alpha fold_exceptional t n ~init ~f =
+    let choose_normal_or_exn_succs node = 
+      let instrs = Procdesc.Node.get_instrs node in
+      let instr_count = (Instrs.count instrs) - 3 in
```

Comment:
Could we have a comment about `- 3` here?

Reply:
Throw exception CFG node actually contains the following three instructions:

![image](https://user-images.githubusercontent.com/63893533/145485884-8aa3d6f5-88ef-4b91-aa05-75b14027a77c.png)

The third from the last instruction is the one we want to verify.

I have added a comment to explain the intention here.

Reply:
Thank you for the explanation.